### PR TITLE
gcc-aarch64 and gcc suffix

### PR DESCRIPTION
**Pull request by jglanz**
Added linux-gcc-aarch64.cmake 
Added suffix support for GCC layouts (I needed it for -6)

### CODE REVIEW

File: compiler/gcc-cross-compile.cmake [hunk 1]
```diff
@@ -40,6 +40,7 @@ if(_is_empty)
   polly_fatal_error("CROSS_COMPILE_TOOLCHAIN_PREFIX not set.")
 endif()
 
+
```
Comment (ruslo) [inline compiler/gcc-cross-compile.cmake]: Cleanup

File: compiler/gcc-cross-compile.cmake [hunk 2]
```diff
@@ -49,6 +50,8 @@ if(_is_empty)
   polly_fatal_error("CROSS_COMPILE_SYSROOT not set.")
 endif()
 
+
+
```
Comment (ruslo) [inline compiler/gcc-cross-compile.cmake]: Cleanup

File: linux-gcc-aarch64.cmake [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+else()
+  set(POLLY_LINUX_GCC_AARCH64_ 1)
+endif()
+
+
+# set system name, this sets the variable CMAKE_CROSSCOMPILING
+set(CMAKE_SYSTEM_NAME Linux)
+
+set(CROSS_COMPILE_TOOLCHAIN_PREFIX aarch64-linux-gnu)
+set(CROSS_COMPILE_TOOLCHAIN_SUFFIX -6)
```
Comment (ruslo) [inline linux-gcc-aarch64.cmake]: Probably should go to the name of toolchain, like `linux-gcc-6-aarch64`.

File: linux-gcc-aarch64.cmake [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  set(SYSROOT $ENV{SYSROOT})
+endif()
+
+set(CROSS_COMPILE_SYSROOT ${SYSROOT})
+
+
+include("${CMAKE_CURRENT_LIST_DIR}/utilities/polly_init.cmake")
+
+polly_init(
+    "Linux / gcc / aarch64 / c++11 support"
```
Comment (ruslo) [inline linux-gcc-aarch64.cmake]: `c++17 support` ?

File: linux-gcc-aarch64.cmake [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+include("${CMAKE_CURRENT_LIST_DIR}/utilities/polly_common.cmake")
+
+
+set(CMAKE_CROSSCOMPILING_EMULATOR qemu-aarch64) # used for try_run calls
+
+include(
+    "${CMAKE_CURRENT_LIST_DIR}/compiler/gcc-cross-compile.cmake"
+)
+
+include("${CMAKE_CURRENT_LIST_DIR}/flags/cxx17.cmake")
```
Comment (ruslo) [inline linux-gcc-aarch64.cmake]: Currently similar toolchains set C++11 by default so if you want C++17 I think we should name toolchain differently, like `linux-gcc-aarch64-cxx17`.

File: linux-gcc-aarch64.cmake [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  set(POLLY_LINUX_GCC_AARCH64_ 1)
+endif()
+
+
+# set system name, this sets the variable CMAKE_CROSSCOMPILING
+set(CMAKE_SYSTEM_NAME Linux)
+
+set(CROSS_COMPILE_TOOLCHAIN_PREFIX aarch64-linux-gnu)
+set(CROSS_COMPILE_TOOLCHAIN_SUFFIX -6)
+set(CROSS_COMPILE_TOOLCHAIN_PATH /usr/bin)
```
Comment (ruslo) [inline linux-gcc-aarch64.cmake]: Currently `CROSS_COMPILE_TOOLCHAIN_PATH` is used in `gcc-musl`/`arm-openwrt-linux-muslgnueabi` and set basing on environment variable that is tied to the toolchain name:
* https://github.com/ruslo/polly/blob/3b5ab745342adbff60ff6f9308bf55aabd19585b/gcc-musl.cmake#L19
* https://github.com/ruslo/polly/blob/3b5ab745342adbff60ff6f9308bf55aabd19585b/arm-openwrt-linux-muslgnueabi.cmake#L37

I think we should follow same approach here:
* Read environment variable `LINUX_GCC_AARCH64_ROOT`
* If not set use default path `/usr/bin` (do we need this?)
* Check path is correct, e.g. `$ENV{GCC_AARCH64_ROOT}/aarch64-linux-gnu-gcc-6` file exist
* If file not found emit error message with clear description of what is expected

File: compiler/gcc-cross-compile.cmake (hunk 1 above)
Comment (ruslo) [reply compiler/gcc-cross-compile.cmake]: Cleanup

File: compiler/gcc-cross-compile.cmake (hunk 2 above)
Comment (ruslo) [reply compiler/gcc-cross-compile.cmake]: Cleanup

**Comment by ruslo** [comment]
Closing inactive pull request.

**Comment by Nadav-Eyesight** [comment]
I don't see any aarch64 toolchains.

**Comment by ruslo** [comment]
@Nadav-Eyesight This pull request was closed without merge because of inactivity. Feel free to continue working on.

**Pull request closed without merging**